langchain/agents/conversational_chat/output_parser.py

```diff
@@ -14,18 +15,29 @@ def get_format_instructions(self) -> str:
 
     def parse(self, text: str) -> Union[AgentAction, AgentFinish]:
         cleaned_output = text.strip()
-        if "```json" in cleaned_output:
-            _, cleaned_output = cleaned_output.split("```json")
-        if "```" in cleaned_output:
-            cleaned_output, _ = cleaned_output.split("```")
-        if cleaned_output.startswith("```json"):
-            cleaned_output = cleaned_output[len("```json") :]
-        if cleaned_output.startswith("```"):
-            cleaned_output = cleaned_output[len("```") :]
-        if cleaned_output.endswith("```"):
-            cleaned_output = cleaned_output[: -len("```")]
-        cleaned_output = cleaned_output.strip()
-        response = json.loads(cleaned_output)
+
+        action_pattern = r'"action":\s*"([^"]*)"'
+        action_input_pattern = r'"action_input":\s*"([^"]*)"'
+
+        action_match = re.search(action_pattern, cleaned_output)
+        action_input_match = re.search(action_input_pattern, cleaned_output)
+
+        try:
+            if action_match is None or action_input_match is None:
+                raise ValueError(
+                    "Failed to parse values from the LLM output: ", cleaned_output
+                )
+
+            action = action_match.group(1)
+            action_input = action_input_match.group(1)
+
+            parsed = {"action": action, "action_input": action_input}
+            parsed_output = json.dumps(parsed)
+
+        except AttributeError:
+            print("Failed to parse LLM output: ", cleaned_output)
+
+        response = json.loads(parsed_output)
```

Comment:
I know this was in part trying to maintain the existing code but it does feel a bit strange to turn the parsed information into a string only to turn it back into json. I think in general what's different about your change and mine is that mine really changes the goal of this function - trying to determine the LLM's intent. So mine doesn't throw any exceptions anymore that aren't handled within this function, and the Agent should always receive an action or finish no matter what.

Reply:
I totally agree with you! I didn't want to change the working behaviour of the function. I mean "turn the parsed information into a string only to turn it back into json".  I think my way also doesn't throw any exceptions anymore but I believe that changing the goal of this function is a better way for future developments.